### DISCUSSION TITLE
(W5500 hardwired TCP/IP embedded Ethernet controller)

### DISCUSSION THREAD

**Discussion by alalal80** [Montecri/GNSSTimeServer#13] [Q&A]
Good morning,
how to connect the module (W5500 hardwired TCP/IP embedded Ethernet controller)
at esp32?
in the current version is it possible to add the second screen.
Sincerely.

**Comment by Montecri** [comment]
Hi!

On the source code, lines 86 to 90, you find the default pin assignment for a regular W5500 module. It is this:

#define MO 23
#define MI 19
#define CL 18
#define CS 5
#define INT 25

So, on the pin module, you will have the pins labeled as MO, MI, CL, CS, and INT. You should connect those to the respective pins on ESP32 per the defines (23, 19, 18, 5 and 25).

Don't forget to uncomment line 39, which will enable W5500 ethernet functionality:

// #define ETHERNET_ENABLED

The current version is not coded for two screens, you will need to pick this specific code from the "Dual-Display" branch, merge and test, or write your own code.

Hope this helped.

Regards,

- Cristiano

**Comment by alalal80** [comment]
Thank you for your answers, have you planned a new version?
Sincerely.

**Comment by Montecri** [comment]
Welcome!

Yes, planning new features. However, the current version is very "fresh"; for the last 6 months, I added several new features, like Ethernet support, enriched webpage, OTA upgrade, on-boot diagnostics, RTOS optimizations, new 3D printable case on OpenSCAD, etc.

I am working now on continuous integration on Github.

Thanks!

- Cristiano

**Comment by alalal80** [comment]
thank you, keep me informed when the new version is made available. Best regards.
